oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/incrementalstore/IncrementalStoreBuilder.java

```diff
@@ -0,0 +1,152 @@
+    private final String initialCheckpoint;
+    private final String finalCheckpoint;
+    private Predicate<String> pathPredicate = path -> true;
+    private Set<String> preferredPathElements = Collections.emptySet();
+    private BlobStore blobStore;
+
+    private final boolean compressionEnabled = Boolean.parseBoolean(System.getProperty(FlatFileNodeStoreBuilder.OAK_INDEXER_USE_ZIP, "true"));
+    private final boolean useLZ4 = Boolean.parseBoolean(System.getProperty(FlatFileNodeStoreBuilder.OAK_INDEXER_USE_LZ4, "false"));
+    private final Compression algorithm = compressionEnabled ? (useLZ4 ? new LZ4Compression() : Compression.GZIP) :
+            Compression.NONE;
+
+
+    /**
+     * System property name for sort strategy. This takes precedence over {@link #INCREMENTAL_SORT_STRATEGY_TYPE}.
+     * Allowed values are the values from enum {@link IncrementalStoreBuilder.IncrementalSortStrategyType}
+     */
+    public static final String INCREMENTAL_SORT_STRATEGY_TYPE = "oak.indexer.incrementalSortStrategyType";
+
+
+    private final String sortStrategyTypeString = System.getProperty(INCREMENTAL_SORT_STRATEGY_TYPE);
+    private IncrementalStoreBuilder.IncrementalSortStrategyType sortStrategyType = sortStrategyTypeString != null
+            ? IncrementalStoreBuilder.IncrementalSortStrategyType.valueOf(sortStrategyTypeString)
+            : IncrementalSortStrategyType.INCREMENTAL_FFS_STORE;
+
+
+    public enum IncrementalSortStrategyType {
+        /**
+         * Incremental store having nodes updated between initial and final checkpoint
+         */
+
+        INCREMENTAL_FFS_STORE
+    }
+
+    public IncrementalStoreBuilder(File workDir, IndexHelper indexHelper,
+                                   @NotNull String initialCheckpoint, @NotNull String finalCheckpoint) {
+        this.workDir = workDir;
+        this.indexHelper = indexHelper;
+        this.initialCheckpoint = Objects.requireNonNull(initialCheckpoint);
+        this.finalCheckpoint = Objects.requireNonNull(finalCheckpoint);
+    }
+
+    public IncrementalStoreBuilder withPreferredPathElements(Set<String> preferredPathElements) {
+        this.preferredPathElements = preferredPathElements;
+        return this;
+    }
+
+    public IncrementalStoreBuilder withSortStrategyType(IncrementalStoreBuilder.IncrementalSortStrategyType sortStrategyType) {
+        this.sortStrategyType = sortStrategyType;
+        return this;
+    }
+
+    public IncrementalStoreBuilder withPathPredicate(Predicate<String> pathPredicate) {
+        this.pathPredicate = pathPredicate;
+        return this;
+    }
+
+    public IncrementalStoreBuilder withBlobStore(BlobStore blobStore) {
+        this.blobStore = blobStore;
+        return this;
+    }
+
+
+    public IncrementalStore build() throws IOException, CompositeException {
+        logFlags();
+        File dir = createStoreDir();
+
+        switch (sortStrategyType) {
+            case INCREMENTAL_FFS_STORE:
```

Comment:
this switch has only one case, I would replace it with a if for better readability
```suggestion
if (Objects.requireNonNull(sortStrategyType) == IncrementalSortStrategyType.INCREMENTAL_FFS_STORE) {
```

Reply:
done